compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/SplitReduction.cpp

```diff
@@ -657,13 +657,25 @@ splitArgmaxReduction(RewriterBase &rewriter, linalg::GenericOp genericOp,
         Value in = args[0];
         Value outVal = args[1];
         Value outIdx = args[2];
-        Value reductionIdx = linalg::IndexOp::create(b, loc, reductionDim + 1);
-        if (outIdx.getType() != reductionIdx.getType())
-          reductionIdx = arith::IndexCastOp::create(b, loc, outIdx.getType(),
-                                                    reductionIdx);
-        Value maxVal = arith::MaximumFOp::create(b, loc, in, outVal);
-        Value cmp = arith::CmpFOp::create(b, loc, arith::CmpFPredicate::OGT, in,
-                                          outVal);
+        Value reductionIdx = b.create<linalg::IndexOp>(loc, reductionDim + 1);
+        // Ensure outIdx and reductionIdx have the same type for subsequent operations.
+        if (outIdx.getType() != reductionIdx.getType()) {
+            reductionIdx =
+                b.create<arith::IndexCastOp>(loc, outIdx.getType(), reductionIdx);
+        }
+        Value inCast = in;
+        if (in.getType() != outVal.getType()) {
+          if (outVal.getType().getIntOrFloatBitWidth() >
+              in.getType().getIntOrFloatBitWidth()) {
+            inCast = b.create<arith::ExtFOp>(loc, outVal.getType(), in);
+          } else {
+            inCast = b.create<arith::TruncFOp>(loc, outVal.getType(), in);
+          }
+        }
```

Comment:
nit: This could just be:
```suggestion
        if (outVal.getType().getIntOrFloatBitWidth() >
            in.getType().getIntOrFloatBitWidth()) {
          inCast = b.create<arith::ExtFOp>(loc, outVal.getType(), in);
        } else if (outVal.getType().getIntOrFloatBitWidth() <
            in.getType().getIntOrFloatBitWidth()) {
          inCast = b.create<arith::TruncFOp>(loc, outVal.getType(), in);
        }
```